### PR TITLE
blog : Sending and receiving Telegram messages with Node-RED

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Related Issue(s)
https://github.com/FlowFuse/customer/issues/187

Closes https://github.com/FlowFuse/website/issues/1924

## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

File: src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+- After that, you'll see the chat interface. Type `/newbot` and press enter to send the command.
+- The bot will ask for a name, which will be the display name of your bot. You can choose any name you like.
+- Next, it will ask for a unique username for your bot. The username should not include spaces and must end with 'bot' for example telegram_bot or telegrambot
+- Once you have entered a correct and unique username, you’ll receive a message confirming that you have successfully created the bot. It will contain the secret access token and a link to initiate your bot.
+- Click on the provided link. You'll be directed to the chat interface between you and the created bot, with a start button at the bottom. Click on that button to activate your bot.
+
+!["screenshot displaying chat interface with start buttom for activating your bot"](./images/sending-telegram-with-node-red-activating-bot.png "screenshot displaying chat interface with start buttom for activating your bot"){data-zoomable}
```
Comment (gdziuba) [inline src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md:31]:
```suggestion
!["screenshot displaying chat interface with start button for activating your bot"](./images/sending-telegram-with-node-red-activating-bot.png "screenshot displaying chat interface with start button for activating your bot"){data-zoomable}
```

File: src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+## Adding environment variables
+
+Utilizing environment variables is crucial while doing any configuration in your application as they contain sensitive information. Environment variables enable you to configure nodes or applications without exposing such information in the flow.json. For more details refer to [Using Environment Variables with Node-RED](https://flowfuse.com/blog/2023/01/environment-variables-in-node-red/).
+
+- Go to Instance settings, and switch to the “Environment” tab.
+- Click the top right “Add variable” button.
+- Add variables for a secret access token and chat ID.
+- Click on the "Save settings" button and restart the instance by clicking on the top "action" button and selecting the "Restart" option.
+
+!["Screenshot displaying flowfuse instance settings for adding enviroment varible"](./images/sending-telegram-with-node-red-flowfue-instance-settings.png "Screenshot displaying flowfuse instance settings for adding enviroment varible"){data-zoomable}
```
Comment (gdziuba) [inline src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md:66]:
```suggestion
!["Screenshot displaying flowfuse instance settings for adding environment varible"](./images/sending-telegram-with-node-red-flowfue-instance-settings.png "Screenshot displaying flowfuse instance settings for adding environment variable"){data-zoomable}
```

File: src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+- Then click on the edit icon next to the Bot input field.
+- Enter your bot name and the environment variable name set for the access token into the Token field, and enter the environment name set for the chat ID into the chatIds input field as shown below, and keep other settings unchanged.
+
+!["Screenshot displaying telegram custom node configuration"](./images/sending-telegram-with-node-red-telegram-node-configuration.png "Screenshot displaying telegram custom node configuration"){data-zoomable}
+
+## Sending a message to telegram
+
+- Drag an Inject node onto the canvas.
+- Drag a Change node onto the canvas. Set msg.payload.type as "message"  To explore other types, refer to the [node readme](https://flows.nodered.org/node/node-red-contrib-telegrambot), set msg.payload.chatId as the environment variable added for the chat ID, and then set msg.payload.content as the message you want to send.
```
Comment (gdziuba) [inline src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md:83]:
```suggestion
- Drag a Change node onto the canvas. Set msg.payload.type as "message."  To explore other types, refer to the [node readme](https://flows.nodered.org/node/node-red-contrib-telegrambot), set msg.payload.chatId as the environment variable added for the chat ID, and then set msg.payload.content as the message you want to send.
```

File: src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+1. Drag a Receiver node onto the canvas.
+2. Click on the node, and ensure you have selected the correct bot configuration that you have added.
+3. For the Receiver node, you'll see two outputs: one for receiving messages from authorized users, and another for receiving messages from unauthorized users.
+4. To add users to your authorized user list, click on the receiver node and click on the edit icon next to the bot input field, then add users separated by commas into the users input field.
+5. Drag two Debug nodes onto the canvas.
+6. Connect the first Debug node’s input to the first output of the Receiver node and the second Debug node’s input to the second output of the Receiver node.
+
+## Deploying the flow
+
+!["Screenshot displaying node-red flow for sending and receving telegram messages"](./images/sending-telegram-with-node-red-flow.png "Screenshot displaying node-red flow for sending and receving telegram messages"){data-zoomable}
```
Comment (gdziuba) [inline src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md:102]:
```suggestion
!["Screenshot displaying node-red flow for sending and receiving telegram messages"](./images/sending-telegram-with-node-red-flow.png "Screenshot displaying node-red flow for sending and receiving telegram messages"){data-zoomable}
```

File: src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+- After that, you’ll receive a message containing your Chat ID and User ID.
+
+### Obtaining your Telegram group Chat ID
+
+- Add `@getmyid_bot` to the group on which you want to send or receive messages using Node-RED.
+- Once the bot joins the group, the bot will send the chat ID.
+
+## Installing custom node
+
+- Switch to the "Install" tab
```
Comment (gdziuba) [inline src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md:53]: Missing a step or two.  Tell the reader to switch over to FlowFuse and log into your node-red instance or something like that.

File: src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+## Obtaining your Telegram Chat ID
+
+Telegram chat ID is a unique identifier for a chat or a group in Telegram, which is required for sending and receiving messages. In this section, you'll learn how to obtain your own and your group's Telegram chat IDs.
+
+### Obtaining your Telegram Chat ID
+
+- Open your Telegram app Click on the top right search icon and type “Get My ID”
+
+!["screenshot displaying searching "Get My ID" bot for obtaining chat id"](./images/sending-telegram-with-node-red-getmyid.png "screenshot displaying searching "Get My ID" bot for obtaining chat id"){data-zoomable}
```
Comment (gdziuba) [inline src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md:41]:
```suggestion
!["screenshot displaying searching &quot;Get My ID&quot; bot for obtaining chat id"](./images/sending-telegram-with-node-red-getmyid.png "screenshot displaying searching &quot;Get My ID&quot; bot for obtaining chat id"){data-zoomable}
```

**Review by gdziuba — CHANGES_REQUESTED** [review]
In stead of bullet points can we do numbers instead since they are steps?

**Comment by sumitshinde-84** [comment]
> In stead of bullet points can we do numbers instead since they are steps?

Apologies for forgetting again. I'll make sure to use numbers instead of bullet points.

**Comment by gdziuba** [comment]
> > In stead of bullet points can we do numbers instead since they are steps?
> 
> Apologies for forgetting again. I'll make sure to use numbers instead of bullet points.

I like the content.

File: src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+title: Sending and receiving Telegram messages with Node-RED
+subtitle: A step-by-step guide to integrating telegram into your node-red application for sending and receiving messages.
+description: Learn to seamlessly integrate Telegram with Node-RED for messaging. Create bots, obtain chat IDs, and send/receive messages, including group messaging. 
+date: 2024-04-08
+authors: ["sumit-shinde"]
+image: 
+tags:
+   - node-red
+   - post
+   - telegram
```
Comment (gdziuba) [inline src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md:11]:
```suggestion
   - telegram
   - bot
```

File: src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
+---
+title: Sending and receiving Telegram messages with Node-RED
+subtitle: A step-by-step guide to integrating telegram into your node-red application for sending and receiving messages.
+description: Learn to seamlessly integrate Telegram with Node-RED for messaging. Create bots, obtain chat IDs, and send/receive messages, including group messaging. 
+date: 2024-04-08
```
Comment (gdziuba) [inline src/blog/2024/04/sending-and-receiving-telegram-messages-with-node-red.md:5]: Need to adjust the date.

**Review by gdziuba — APPROVED** [review]

**Comment by gdziuba** [comment]
It needs art tile, then it is good to go.

**Review by ZJvandeWeg — APPROVED** [review]

**Pull request merged**